#! /bin/sh /usr/share/dpatch/dpatch-run
## sensors-ignorelist.dpatch by Sebastian Harl <sh@tokkee.org>
##
## DP: sensors plugin: Avoid assertion in ignorelist_match () when not
## DP: configured.

@DPATCH@

diff -uN a/src/sensors.c b/src/sensors.c
--- a/src/sensors.c
+++ b/src/sensors.c
@@ -393,7 +393,8 @@
 	sensors_free_features ();
 #endif /* if SENSORS_HAVE_READ */
 
-	ignorelist_free (sensor_list);
+	if (NULL != sensor_list)
+		ignorelist_free (sensor_list);
 }
 
 static void sensors_voltage_write (char *host, char *inst, char *val)
@@ -402,7 +403,7 @@
 	int status;
 
 	/* skip ignored in our config */
-	if (ignorelist_match (sensor_list, inst))
+	if ((NULL != sensor_list) && ignorelist_match (sensor_list, inst))
 		return;
 
 	/* extended sensor naming */
@@ -423,7 +424,7 @@
 	int status;
 
 	/* skip ignored in our config */
-	if (ignorelist_match (sensor_list, inst))
+	if ((NULL != sensor_list) && ignorelist_match (sensor_list, inst))
 		return;
 
 	/* extended sensor naming */
@@ -450,7 +451,7 @@
 		return;
 
 	/* skip ignored in our config */
-	if (ignorelist_match (sensor_list, inst))
+	if ((NULL != sensor_list) && ignorelist_match (sensor_list, inst))
 		return;
 
 	if (snprintf (buf, BUFSIZE, "%u:%.3f", (unsigned int) curtime,
